### ISSUE TITLE
Bug in metadata

### ISSUE THREAD

**Issue by agriggio** [artpixls/ART#99]
**[Original report](https://bitbucket.org/agriggio/art/issues/99) by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

small bug on the metadata : 

I noticed two problems:

1 - When I export a raw file \(Nef\) in jpg, the notation is no longer present in the metadata of the jpg file. I manage my metadata with Digikam upstream of the export to do it only once. All other metadata looks like it is properly exported \(title, description, copyright, keywords, color labels, …\) read from Digikam.

2 - On the other hand, I noticed that the exported metadata are quite corrupted when read with some software. They don’t seem to see some of them such as keywords, color labels, titles, descriptions, either they appear truncated or not at all. I tested for example with View NX-i and Xnview MP. If I reassign the metadata with Digikam, the data is again properly accessible in View NX-I. It is only after the export to art that there seems to be a “corruption” of the data.

I add the following attachments :

\_the NEF file

\_the sidecar xmp and arp files

\_jpg export

\_screenshots from Digikam, Xnview MP and View NX-i software

‌

I run art 1.3 on ubuntu 20.04 \(win 10 for View NX-I\).

the attachment file link : [https://mega.nz/file/XIxmxCDQ#OtQkAWUpAVrac-iOBen0jKNYvu5oNgHN\_kYWx\_9px\_w](https://mega.nz/file/XIxmxCDQ#OtQkAWUpAVrac-iOBen0jKNYvu5oNgHN_kYWx_9px_w)

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Hi, I don’t understand what is wrong and what is expected, sorry… can you be more specific? Also, what is “notation” exactly? I checked with exiftool but the metadata seems there. However, here are a couple of suggestions:

1. try updating to the latest version from git master, as there were some metadata bugs fixed after 1.3 was released
2. if you use an external tool to manage your metadata, it’s better to use “copy unchanged” rather than “apply modifications”, like this:

![](https://bitbucket.org/repo/9pArXyE/images/4168780469-Screenshot%20from%202020-06-02%2015.24.59.jpeg)
‌

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

I've just checked but even changing to "copy as is" doesn't change the problem.

If I was not clear enough sorry, I explain myself by taking again the operating mode:  
1 - I complete title, description, key words, color label and notation \(the stars\) on the raw file \(writing a sidecar \*.nef.xmp file\) in Digikam.  
2 - I make the necessary settings on this raw file in ART then export as jpg \(with the option "copy as is" or "apply changes" whatever it seems not to change the problem\)

3 - When I select the freshly exported jpg file with Digikam, the metadata appears correctly, but not in other software. In some the data is partial \(View NX-I, Gthumb\) in others it lacks the rating \(the stars\) \(Xnview MP, View NX-I, Gthumb\).

4 - If I rewrite the same metadata as the raw in the same jpg file using Digikam then the metadata appear correctly in all the software.

Until now I was only using Digikam, so I wasn't aware of the problem.

Hoping to have been more understandable.

‌

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

I just checked no change with the latest version of the appimage \(1.3-104\).

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

I just checked with XnviewMP \(on Linux\), and indeed it doesn’t pick up the rating. However, the rating is definitely there \(in `Exif.Image.Rating`\), it’s just that XnviewMP does’t read that \(source: [https://newsgroup.xnview.com/viewtopic.php?t=38150](https://newsgroup.xnview.com/viewtopic.php?t=38150)l\). My understanding is that XnviewMP reads ratings from XMP tags. I’ll see if I can use that. But overall, I don’t think this is a proper bug, it’s just that the two applications behave differently…

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

I may be wrong of course, I don't have your expertise but look at point 4. I explain, if I reassign \(rewrite\) the same metadata with the command "write metadata the selected files" they appear completely in gthumb, Xnview MP and View NX-I. It is only when exporting that you "lose" something. [Gif demo](https://mega.nz/file/jZwQFIqS#y940JpbjoNdMqkJ-WUJi-dbdYu4pYNEGh1V6Awe4iW0)

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

yes, I understand, but are you sure that digikam is not applying some “magic” here? IIRC it has some smart logic to keep some metadata fields \(comments, tags, ratings\) in different formats \(see “Preferences → Metadata → Advanced”\). ART is very basic, it doesn’t have such things – and they are not planned.

That said, I am looking again to see if there’s something I am missing.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

I made a couple of changes that try to better preserve metadata in multiple places \(exif/iptc/xmp\). However, for maximum interoperability with other apps that use xmp sidecars \(such as XnviewMP\), I suggest you use the following settings in both ART and digiKam:

![](https://bitbucket.org/repo/9pArXyE/images/3244792567-Screenshot%20from%202020-06-03%2011.49.20.jpeg)
![](https://bitbucket.org/repo/9pArXyE/images/393026713-Screenshot%20from%202020-06-03%2011.50.08.jpeg)
‌

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

I've just tested it, the modification doesn't change anything:  
\_ in Xnview MP, the rating is still missing, all the other metadata is present.  
\_ in Gthumb and View NX-I: only one keyword appears, and the rating is back to 0 star.

From what I could see, only IPTC \(keywords and rating\) are modified when exporting the file in ART, not the xmp which they are always well coded. Gthumb and View NX-I read keywords and rating in IPTC and not in XMP.

If I reassign the metadata on the file exported with Digikam, all the softwares show me all the metadata properly.

When testing on a TIFF file it's the same thing.

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

in fact there is a change but in digikam, the rating doesn't appear anymore in the jpg file, whereas with the previous options it kept it.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

it’s clear that art and the other viewers use different conventions. What is not clear to me is that this is a bug in art. Art doesn’t use iptc for rating, but xmp. From my point of view, it works as advertised, so I’m afraid there's not much I can do, sorry…

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

did you simply update art or did you also change the preferences about the xmp sidecars? I'm trying to understand here what is going on. If there's a regression wrt the interoperability with digikam I'll fix that, but I won't fix any incompatibility with 3rd party apps – the bottom line is this: if you need more-than-basic metadata support, don't rely on art, it's not meant for that

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

I understand your point of view and I am already very pleased with ART's capabilities. On the other hand, I never modify any metadata in ART, I only manage it in Digikam which does it very well, I just notice that the IPTC metadata are not the ones I indicated in Digikam. But maybe the problem comes from Digikam which encodes IPTC differently in jpg and sidecar files? I will look on my side if it doesn't come from there.

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

After some tests the rating is well encoded in digikam in the XMP of the NEF sidecar file, and it is well read in Xnview. But on export with the option "copy unchanged" in ART, the classification is 0 in the XMP metadata in the jpg file.

![](https://bitbucket.org/repo/9pArXyE/images/3813566519-Peek%2003-06-2020%2019-57.gif)
‌

“did you simply update art or did you also change the preferences about the xmp sidecars?” I change the preferences in art and I ran some tests on some files that hadn't been modified by following the same procedure as described above.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

thanks for the additional info. I will take another look

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

I just tried. See the video here:

[https://filebin.net/3kquabgq0gdwh7v2/Peek\_2020-06-03\_21-43.mp4?t=cjz84qpl](https://filebin.net/3kquabgq0gdwh7v2/Peek_2020-06-03_21-43.mp4?t=cjz84qpl)

To me, everything seems to work fine… so I suspect there might be some configuration issue on your side? Could that be possible?

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

Thanks for your time on my problem. For the record, I have no idea where to look. I deleted the settings folder of ART, followed your recommendations for metadata settings for ART and Digikam, try to install ART from PPA \(1.2\) and not using APPIMAGE, it doesn't change anything. As you say and you showed me, the problem comes from my house. I will test in a virtual machine with a fresh installation or another distribution.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

did you try resetting the other apps too? E.g. maybe XnviewMP has a cache folder and/or somehow “broken” config settings?

**Comment by agriggio** [comment]
**Original comment by Laurent (Bitbucket: [Laurent](https://bitbucket.org/Laurent), GitHub: [Laurent](https://github.com/Laurent)).**

----------------------------------------

Yes I re-initialized all of them, I even tested with other versions of ART \(the last appimage\), of digikam, it may have nothing to do but it works with darktable, not with rawtherapee but I think it doesn't work already.  
I'm a bit lost, even if it's not in the most important features, for the moment I'll integrate the metadata after the export, it will get around the problem.
